### PR TITLE
Update form.tsx

### PR DESCRIPTION
**Pull request by saro9588**
### Issues

1. Check doesn't exist if a user try's to create a new account with an email that an account already exists for. 
2. Upon successful creation of a new unique account, the user doesn't automatically gets directed to the login page. 
3. Both on Login and Register/Create Account pages, the input fields are not required. 

### Solutions

1. Render an error message that says "An error occurred. Please try again later." when a user try's to create a new account with an existing account email. 
  Added QueryResult in auth/register/route.js to check if an email already exists. 
2. Update router.push() in register/form.tsx to redirect the user to the login page.
3. Add required attribute to the <input> html elements in both register & login forms.

### CODE REVIEW

**Review by SevanBadal — CHANGES_REQUESTED** [review]
Please add a detailed description to this PR. 'll review afterwards! Thanks @saro9588

File: app/api/auth/register/route.ts [hunk 1]
```diff
@@ -5,6 +5,17 @@ import { NextResponse } from "next/server";
...
     try {
         const {email, password} = await request.json()
+        // Check if the email already exists in the database
+        const data: QueryResult = await sql`
+          SELECT email FROM users WHERE email = ${email}
+        `;
+        const userExists = data.rows.length > 0;
+        if (userExists) {
+          return NextResponse.json(
+            { error: "User with provided email already exists." },
```
Comment (SevanBadal) [inline app/api/auth/register/route.ts:15]: Responding with a message that indicates a user is already created with a given email exposes us to [user enumeration attacks](https://wiki.owasp.org/index.php/Testing_for_User_Enumeration_and_Guessable_User_Account_(OWASP-AT-002)).

File: app/api/auth/register/route.ts [hunk 2]
```diff
@@ -5,6 +5,17 @@ import { NextResponse } from "next/server";
 export async function POST(request: Request) {
     try {
         const {email, password} = await request.json()
+        // Check if the email already exists in the database
+        const data: QueryResult = await sql`
```
[the remark this answers is not in the document]
Comment (saro9588) [reply app/api/auth/register/route.ts:9]: done

**Review by SevanBadal — APPROVED** [review]
Looks good! Thanks, @saro9588

**Pull request merged**